rclcpp_components/test/benchmark/benchmark_components.cpp

```diff
@@ -0,0 +1,117 @@
+
+#include <rcutils/logging.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp_components/component_manager.hpp"
+
+class ComponentTest : public benchmark::Fixture
+{
+public:
+  ComponentTest()
+  : component_manager_name("my_manager")
+  {
+  }
+
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Woverloaded-virtual"
+#endif
+  void SetUp(benchmark::State &) override
+  {
+    rcutils_logging_set_default_logger_level(RCUTILS_LOG_SEVERITY_WARN);
+
+    context = std::make_shared<rclcpp::Context>();
+    context->init(0, nullptr, rclcpp::InitOptions().auto_initialize_logging(false));
+
+    rclcpp::ExecutorOptions exec_options;
+    exec_options.context = context;
+
+    executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>(exec_options);
+
+    manager = std::make_shared<rclcpp_components::ComponentManager>(
+      executor, component_manager_name, rclcpp::NodeOptions().context(context));
+    executor->add_node(manager);
+  }
+
+  void TearDown(benchmark::State &) override
+  {
+    context->shutdown("Test is complete");
+
+    manager.reset();
+    executor.reset();
+    context.reset();
+  }
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
+  const std::string component_manager_name;
+
+protected:
+  rclcpp::Context::SharedPtr context;
+  rclcpp::executors::SingleThreadedExecutor::SharedPtr executor;
+  std::shared_ptr<rclcpp_components::ComponentManager> manager;
+};
+
+BENCHMARK_F(ComponentTest, get_component_resources)(benchmark::State & state)
+{
+  for (auto _ : state) {
+    std::vector<rclcpp_components::ComponentManager::ComponentResource> resources =
+      manager->get_component_resources("rclcpp_components");
+    if (resources.size() != 3) {
+      state.SkipWithError("Wrong number of components found");
+      break;
+    }
+  }
+}
+
+BENCHMARK_F(ComponentTest, create_component_factory)(benchmark::State & state)
+{
+  const std::vector<rclcpp_components::ComponentManager::ComponentResource> resources =
+    manager->get_component_resources("rclcpp_components");
+  if (resources.size() != 3) {
+    state.SkipWithError("Wrong number of components found");
+    return;
+  }
+
+  for (auto _ : state) {
+    manager->create_component_factory(resources[0]).reset();
+  }
+}
+
+BENCHMARK_F(ComponentTest, create_node_instance)(benchmark::State & state)
+{
+  const std::vector<rclcpp_components::ComponentManager::ComponentResource> resources =
+    manager->get_component_resources("rclcpp_components");
+  if (resources.size() != 3) {
+    state.SkipWithError("Wrong number of components found");
+    return;
+  }
+
+  const std::shared_ptr<rclcpp_components::NodeFactory> factory =
+    manager->create_component_factory(resources[0]);
+
+  const rclcpp::NodeOptions options = rclcpp::NodeOptions().context(context);
+
+  for (auto _ : state) {
+    rclcpp_components::NodeInstanceWrapper node = factory->create_node_instance(options);
```

Comment:
node here could probably get `DoNotOptimize()/ClobberMemory()` since nothing is being done with the instance.

Reply:
Okay, 42f5919